specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2021-09-01-preview/grafana.json

```diff
@@ -0,0 +1,748 @@
+          "type": "string",
+          "readOnly": true
+        },
+        "sku": {
+          "description": "The Sku of the grafana resource.",
+          "$ref": "#/definitions/ResourceSku"
+        },
+        "properties": {
+          "description": "Properties specific to the grafana resource.",
+          "$ref": "#/definitions/GrafanaResourceProperties"
+        },
+        "identity": {
+          "description": "The managed identity of the grafana resource.",
+          "$ref": "#/definitions/ManagedIdentity"
+        },
+        "systemData": {
+          "description": "The system meta data relating to this grafana resource.",
+          "$ref": "#/definitions/SystemData"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "location": {
+          "type": "string"
+        }
+      }
+    },
+    "GrafanaResourceListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GrafanaResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "GrafanaResourceProperties": {
+      "type": "object",
+      "description": "Properties specific to the grafana resource.",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "grafanaVersion": {
+          "description": "The Grafana software version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "endpoint": {
+          "description": "The endpoint of the Grafana instance.",
+          "type": "string",
+          "readOnly": true
+        },
+        "zoneRedundancy": {
+          "$ref": "#/definitions/ZoneRedundancy"
+        }
+      }
+    },
+    "GrafanaResourcePropertiesUpdateParameters": {
+      "type": "object",
+      "properties": {
+        "zoneRedundancy": {
+          "description": "The high availability mode of the grafana resource.",
+          "$ref": "#/definitions/ZoneRedundancy"
+        }
+      }
+    },
+    "GrafanaResourceUpdateParameters": {
+      "type": "object",
+      "description": "The parameters for a PATCH request to a grafana resource.",
+      "properties": {
+        "identity": {
+          "description": "The managed identity of the grafana resource.",
+          "$ref": "#/definitions/ManagedIdentity"
+        },
+        "tags": {
+          "type": "object",
+          "description": "The new tags of the grafana resource.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "location": {
+          "description": "The location of the grafana resource.",
+          "type": "string"
+        },
```

Comment:
you cannot update the location in a PATCH command

Reply:
Hi @raosuhas , the context is there is a bug in Portal side which forces us to use PATCH as a workaround. Here are some discussions with Jeffrey Richter and Arthur Ning about it before: https://github.com/Azure/azure-rest-api-specs-pr/pull/5474#issuecomment-969136518

Reply:
After double check, the Portal bug is fixed. I've removed the `location` from PATCH.